# Homework Assignment - Lesson 1: Introduction to R

**Student Name:** Ramiro Cavazos  
**Date:** September 2, 2025  
**Course:** Data Management  

---

## Instructions

Complete all the tasks below by adding your R code in the code cells and your written responses in markdown cells. Make sure to run all cells to show your output.

**📋 SUBMISSION**: When you're done, see [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for complete submission instructions.

---

## Part 1: Setting Up Your Environment and Importing Data

### 1.1 Working Directory

Check your current working directory using `getwd()`. You should be in the main repository folder.

In [1]:
# Check current working directory
getwd()

[1] "/workspaces/assignment-1-therocketter5/assignment/Homework"

### 1.2 Package Loading

Load the required packages: `tidyverse` and `readxl`

In [2]:
# Load required packages
library(tidyverse)
library(readxl)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.1.0     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


### 1.3 Data Import - CSV

Import the `sales_data.csv` file from the `data/` folder into a data frame named `sales_df`.

In [3]:
# Import sales data
sales_df <- read_csv("/workspaces/assignment-1-therocketter5/data/sales_data.csv")
print("Sales data imported successfully!")

Rows: 100 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): Product, Region
dbl  (3): TransactionID, Amount, Quantity
date (1): Date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] "Sales data imported successfully!"


### 1.4 Data Import - Excel

Import both sheets from the `customer_feedback.xlsx` file.

In [5]:
# Import ratings data from Excel
ratings_df <- read_excel("/workspaces/assignment-1-therocketter5/data/customer_feedback.xlsx", sheet = "ratings")

# Import comments data from Excel  
comments_df <- read_excel("/workspaces/assignment-1-therocketter5/data/customer_feedback.xlsx", sheet = "customer_feedback")

print("Excel data imported successfully!")

[1] "Excel data imported successfully!"


---

## Part 2: Basic Data Inspection

For each of the three data frames you imported, perform the following inspection tasks. Add both your R code and a brief interpretation of what you observe.

### 2.1 Sales Data Inspection

In [7]:
# First 10 rows of sales data
print("First 10 rows of sales_df:")
head(sales_df, 10)
# the Sales_df sheet has columbs of transactionID, Date, Product, Amount, Quantity and Region

[1] "First 10 rows of sales_df:"


TransactionID Date       Product  Amount  Quantity Region
1   1            2023-01-27 Keyboard 1039.05 1        South 
2   2            2023-05-18 Tablet    597.55 5        East  
3   3            2023-02-14 Keyboard 1539.48 1        West  
4   4            2023-07-28 Mouse     261.66 1        South 
5   5            2023-05-17 Monitor  1047.73 1        South 
6   6            2023-02-25 Keyboard 1824.05 2        East  
7   7            2023-01-20 Monitor   475.83 1        North 
8   8            2023-07-14 Mouse     758.05 5        North 
9   9            2023-10-02 Mouse    1717.20 4        South 
10 10            2023-10-15 Laptop   1437.21 3        North

In [9]:
# Structure of sales data
print("Structure of sales_df:")
str(sales_df)
# Sales Data has three number columns (TransactionID, Amount and Quantity), Two Character Columns (Region and Product) and the Date Column in Date formatting

[1] "Structure of sales_df:"
spc_tbl_ [100 × 6] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ TransactionID: num [1:100] 1 2 3 4 5 6 7 8 9 10 ...
 $ Date         : Date[1:100], format: "2023-01-27" "2023-05-18" ...
 $ Product      : chr [1:100] "Keyboard" "Tablet" "Keyboard" "Mouse" ...
 $ Amount       : num [1:100] 1039 598 1539 262 1048 ...
 $ Quantity     : num [1:100] 1 5 1 1 1 2 1 5 4 3 ...
 $ Region       : chr [1:100] "South" "East" "West" "South" ...
 - attr(*, "spec")=
  .. cols(
  ..   TransactionID = col_double(),
  ..   Date = col_date(format = ""),
  ..   Product = col_character(),
  ..   Amount = col_double(),
  ..   Quantity = col_double(),
  ..   Region = col_character()
  .. )
 - attr(*, "problems")=<externalptr> 


In [4]:
# Summary statistics for sales data
print("Summary statistics for sales_df:")
summary(sales_df)

[1] "Summary statistics for sales_df:"


 TransactionID         Date              Product              Amount       
 Min.   :  1.00   Min.   :2023-01-01   Length:100         Min.   :  58.06  
 1st Qu.: 25.75   1st Qu.:2023-03-20   Class :character   1st Qu.: 520.79  
 Median : 50.50   Median :2023-07-03   Mode  :character   Median : 960.75  
 Mean   : 50.50   Mean   :2023-06-27                      Mean   : 962.71  
 3rd Qu.: 75.25   3rd Qu.:2023-09-25                      3rd Qu.:1384.63  
 Max.   :100.00   Max.   :2023-12-26                      Max.   :1985.60  
    Quantity       Region         
 Min.   :1.00   Length:100        
 1st Qu.:1.00   Class :character  
 Median :3.00   Mode  :character  
 Mean   :2.69                     
 3rd Qu.:4.00                     
 Max.   :5.00                     

**Your observations about sales_df:**

There are six columns, TransactionID, Date, Product, Amount , Quantity and Region.

The types of data are as follows:Transaction ID, Amount and Quantity are numbers, Amount and Region are characters and Date is a Date.

There are 100 Transaction ID's ranging from 1 to 100, Dates from Jan 1st 2023 to Dec 26 2023
There are 100 Products with amounts ranging from 58.06 to 1985.60, Quantities from 1 to 5 and 100 Regions.

### 2.2 Ratings Data Inspection

In [14]:
# TODO: Add your code here for ratings_df inspection
# - First 10 rows
print(head(ratings_df))
# - Structure  
print(paste("Structure:", str(ratings_df)))
# - Summary statistics
summary(ratings_df)

# A tibble: 6 × 5
  CustomerID ProductRating ServiceRating OverallSatisfaction RecommendationScore
       <dbl>         <dbl>         <dbl>               <dbl>               <dbl>
1          1             1             4                   3                   2
2          2             3             2                   4                   7
3          3             1             5                   5                   1
4          4             1             2                   3                   6
5          5             5             1                   2                   2
6          6             4             2                   2                   6
tibble [50 × 5] (S3: tbl_df/tbl/data.frame)
 $ CustomerID         : num [1:50] 1 2 3 4 5 6 7 8 9 10 ...
 $ ProductRating      : num [1:50] 1 3 1 1 5 4 3 5 4 4 ...
 $ ServiceRating      : num [1:50] 4 2 5 2 1 2 5 2 1 1 ...
 $ OverallSatisfaction: num [1:50] 3 4 5 3 2 2 5 5 3 2 ...
 $ RecommendationScore: num [1:50] 2 7 1 6 2 6 5 10 6

   CustomerID    ProductRating ServiceRating  OverallSatisfaction
 Min.   : 1.00   Min.   :1.0   Min.   :1.00   Min.   :1.00       
 1st Qu.:13.25   1st Qu.:2.0   1st Qu.:1.00   1st Qu.:2.00       
 Median :25.50   Median :3.0   Median :3.00   Median :3.00       
 Mean   :25.50   Mean   :3.2   Mean   :2.94   Mean   :2.92       
 3rd Qu.:37.75   3rd Qu.:5.0   3rd Qu.:5.00   3rd Qu.:4.00       
 Max.   :50.00   Max.   :5.0   Max.   :5.00   Max.   :5.00       
 RecommendationScore
 Min.   : 0.0       
 1st Qu.: 1.0       
 Median : 4.5       
 Mean   : 4.2       
 3rd Qu.: 7.0       
 Max.   :10.0       

**Your observations about ratings_df:**

Looking at the head of the list shows 5 categories of Customer ID, ProductRating, ServiceRating, OverallSatisfaction, RecommendationScore.
All of these are categories of numbers, CustomerID ranges from 1 to 50, ProductRating ranges from 1 to 5 and averages a 3.2
ServiceRating also ranges from 1 to 5 and averages a 2.94, Overall Satisfaction ranges from 1 to 5 and averages a 2.92
Finally, RecommendationScore ranges from 0 to 10 and averages a 4.2

### 2.3 Comments Data Inspection

In [15]:
# TODO: Add your code here for comments_df inspection
# - First 10 rows
print("First 10 Rows:")
head(comments_df)
# - Structure
print("Structure:")
structure(comments_df)
# - Summary statistics
print("Summary:")
summary(comments_df)

[1] "First 10 Rows:"


FeedbackID CustomerID Feedback_Text                    Contact_Info          
1 1          12         Highly recommend this item       bob.wilson@test.org   
2 2          40         Excellent service                555-123-4567          
3 3          34         Poor quality control             jane_smith@company.com
4 4           1         average product, nothing special jane_smith@company.com
5 5          47         AMAZING customer support!!!      555-123-4567          
6 6          13         AMAZING customer support!!!      john.doe@email.com    
  Feedback_Date
1 2024-02-23   
2 2024-01-21   
3 2023-09-02   
4 2023-08-21   
5 2023-04-24   
6 2023-04-16

[1] "Structure:"


FeedbackID CustomerID Feedback_Text                      
1    1         12         Highly recommend this item         
2    2         40         Excellent service                  
3    3         34         Poor quality control               
4    4          1         average product, nothing special   
5    5         47         AMAZING customer support!!!        
6    6         13         AMAZING customer support!!!        
7    7         13         average product, nothing special   
8    8         37         good VALUE for money               
9    9         49         Highly recommend this item         
10  10         23         Highly recommend this item         
11  11         24         Excellent service                  
12  12         35         Great product! Love it!            
13  13          3         Poor quality control               
14  14         20         terrible quality, very disappointed
15  15         38         Highly recommend this item         
16  16         15         Excellent service                  
17  17         45         average product, nothing special   
18  18          5         good VALUE for money               
19  19         29         Bad experience with delivery       
20  20         27         Bad experience with delivery       
21  21         21         terrible quality, very disappointed
22  22         16         Excellent service                  
23  23          2         average product, nothing special   
24  24         17         Highly recommend this item         
25  25         47         terrible quality, very disappointed
26  26          9         AMAZING customer support!!!        
27  27          8         terrible quality, very disappointed
28  28          8         good VALUE for money               
29  29         46         Great product! Love it!            
30  30         11         AMAZING customer support!!!        
⋮   ⋮          ⋮          ⋮                                  
71   71        41         terrible quality, very disappointed
72   72        12         good VALUE for money               
73   73        12         Outstanding performance            
74   74        16         Poor quality control               
75   75        27         average product, nothing special   
76   76        30         average product, nothing special   
77   77        38         Great product! Love it!            
78   78        26         average product, nothing special   
79   79        17         AMAZING customer support!!!        
80   80        26         good VALUE for money               
81   81        22         Highly recommend this item         
82   82         9         average product, nothing special   
83   83        37         terrible quality, very disappointed
84   84        35         Poor quality control               
85   85        49         Outstanding performance            
86   86        18         terrible quality, very disappointed
87   87        20         average product, nothing special   
88   88        49         average product, nothing special   
89   89        23         Excellent service                  
90   90        48         AMAZING customer support!!!        
91   91        15         Excellent service                  
92   92         5         Bad experience with delivery       
93   93        46         Outstanding performance            
94   94        35         Bad experience with delivery       
95   95        34         AMAZING customer support!!!        
96   96         1         Poor quality control               
97   97        40         Excellent service                  
98   98        30         terrible quality, very disappointed
99   99        38         good VALUE for money               
100 100        50         Poor quality control               
    Contact_Info           Feedback_Date
1   bob.wilson@test.org    2024-02-23   
2   555-123-4567           2024-01-21   
3   jane_smith@company.com 2023-09-02

[1] "Summary:"


   FeedbackID       CustomerID    Feedback_Text      Contact_Info      
 Min.   :  1.00   Min.   : 1.00   Length:100         Length:100        
 1st Qu.: 25.75   1st Qu.:15.00   Class :character   Class :character  
 Median : 50.50   Median :26.00   Mode  :character   Mode  :character  
 Mean   : 50.50   Mean   :26.27                                        
 3rd Qu.: 75.25   3rd Qu.:38.00                                        
 Max.   :100.00   Max.   :50.00                                        
 Feedback_Date                
 Min.   :2023-01-03 00:00:00  
 1st Qu.:2023-04-23 18:00:00  
 Median :2023-08-24 12:00:00  
 Mean   :2023-08-22 07:26:24  
 3rd Qu.:2023-12-07 06:00:00  
 Max.   :2024-03-30 00:00:00  

**Your observations about comments_df:**

Comments_DF features 5 categories, FeedbackID a number, CustomerID a number, feedback_text which are character values, contact_info which are character values and feedback_date which are date values.

The entire dataset is mostly comprised of characters with multiple missing entries in the contact_info category which is marked "invalid-email" and also features a mix of email and phone numbers. 

Feedback ID goes from 1 to 100, Customer ID goes from 1 to 50, there are 100 entries each of feedback_text and contact_info and the feedback_date goes from Jan 1 2023 to March 30 2024

**Reflection Question 1:** What did you observe about the structure and content of the data files you loaded? What insights can you share about the quality and organization of the data?

**Your Answer:**

The data sets are overall well organized, however inherit issues exist when it comes to the Comments data frame which causes it to require a lot of work and data cleaning for it to be useful in data analysis. The structure is sound for all three and cater to the types of data being collected.

### Question 1: Data Types Analysis

Based on your inspection of `sales_df`, what are the data types of the `Date` and `Amount` columns? Are these data types appropriate for typical business analytics tasks involving sales data? Explain why or why not.

**Your Answer:**

the Date column is stored as a date in the format YYYY-MM-DD which allows it to be treated like a date, which makes it easily sortable, allowing for month-by-month trends to be created. The amount column is stored as a double number, which allows it to be manipulated and adjusted (in conjunction with the quantity column) this makes these types appropriate for sales data.

### Question 2: Data Quality Assessment

Looking at all three datasets, what potential data quality issues do you notice? Consider missing values, data types, and any unusual patterns.

**Your Answer:**

For the Sales_df sheet it is mostly fine, data is filled in accurately, the content seems sound. Ratings_DF is purely catogrized ratings from 1 to 5 or 1 to 10 which makes it easy for analysis, however these ratings are sorted by customer id which requires more context especially when it comes to product rating, meaning that the use of the sales and ratings sheets to do analysis. Finally the comments_df, this sheet is problematic with multiple fields of missing data, data only in plain text. It's use is limited by the stakeholder's value of text-based scrutiny of their products over the ratings sheet.

### Question 3: Analysis Readiness

Which of the three datasets appears most ready for analysis, and which would require the most preprocessing? Justify your answer.

**Your Answer:**

I think that the ratings_df is the most ready for analysis due to it's large amounts of number data only needing some more context (such as a product column) to gain valuable insight. The most processing needs to happen for the comments_df dataset to be useful to the end user. It would require signifcant data cleanup (such as the creation of a phone column and an email column, and the removal of entries that fail to provide either), this set is most useful for a company who wants to do targeted analysis or a product study with a group of customers.

---

## Submission Checklist

Before submitting, make sure you have:

- [ ] Completed all code cells in Parts 1 and 2
- [ ] Run all cells to show output
- [ ] Added your observations for each dataset
- [ ] Answered all reflection questions in Part 3
- [ ] Added your name and date at the top
- [ ] Saved the notebook
- [ ] Committed and pushed your changes to GitHub

**Great work completing your first R data analysis assignment! 🎉**

---

## 🚀 Ready to Submit?

### Easy Submission Steps (No Command Line Required!):

1. **Save this notebook** (Ctrl+S or File → Save)

2. **Use VS Code Source Control**:
   - Click the **Source Control** icon in the left sidebar (tree branch symbol)
   - Click the **"+"** button next to your notebook file
   - Type a message: `Submit homework 1 - [Your Name]`
   - Click **"Commit"** 
   - Click **"Sync Changes"** or **"Push"**

3. **Verify on GitHub**: Go to your repository online and confirm your notebook appears with your completed work

**📖 Need help?** See [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for detailed instructions.

**🎉 Congratulations on completing your first R data analysis assignment!**